### PR TITLE
Add checkout to version in install linux distro

### LINKED ISSUE #1667: Missing checkout to correct branch for installing from source 1.6.6
For the 'Installing' instructions the page https://docs.buildstream.build/1.6/install_linux_distro.html#installing-from-source is missing `git checkout bst-1` before `pip3 install --user -e .`

### PR DESCRIPTION
**Pull request by LawrenceHunter**
Add command to checkout to version tag when installing buildstream. Resolves #1667

### CODE REVIEW

File: doc/source/install_linux_distro.rst [hunk 1]
```diff
@@ -127,6 +127,7 @@ git repository and install it as a regular user::
 
   git clone https://gitlab.com/BuildStream/buildstream.git
   cd buildstream
+  git checkout <VERSION>
```
Comment (jjardon) [inline doc/source/install_linux_distro.rst]: Thanks @LawrenceHunter1 , unless this is replaced at build time with the latest stable tag, I think is better to change this to "git checkout bst-1"

Comment (LawrenceHunter) [reply doc/source/install_linux_distro.rst]: That works better for the latest version but it wouldn't resolve the same issue for example 1.4.2 which is also missing the branching command. This is based on my assumption when reading the docs you're reading the latest version, as the Installing from source page does not vary between which version is selected to install.

Comment (jjardon) [reply doc/source/install_linux_distro.rst]: I think you are right; but then you need to explain what "version" is (tag in the git repo), so there is no confusion

**Comment by gtristan** [comment]
Looks good to me !

Thanks for your contribution.

**Pull request merged**